volterra/resource_volterra_api_credential.go

```diff
@@ -169,9 +170,24 @@ func resourceVolterraAPICredentialUpdate(d *schema.ResourceData, meta interface{
 }
 
 func resourceVolterraAPICredentialDelete(d *schema.ResourceData, meta interface{}) error {
-	// cannot delete api credential object
+	client := meta.(*APIClient)
+	apiCredReq := &ves_io_schema_api_credential.GetRequest{
+		Name:      d.Id(),
```

Comment:
If you take care of my above comment, then you probably have to change this as well